libres/lib/analysis/modules/ies_enkf_data.c

```diff
@@ -161,19 +161,28 @@ void ies_enkf_data_store_initialE(ies_enkf_data_type *data,
         data->E = matrix_alloc(obs_size_msk, ens_size_msk);
         matrix_set(data->E, -999.9);
         int m = 0;
-        for (int i = 0; i < obs_size_msk; i++) {
-            if (bool_vector_iget(data->obs_mask0, i)) {
-                matrix_copy_row(data->E, E0, i, m);
+        for (int iobs = 0; iobs < obs_size_msk; iobs++) {
+            if (bool_vector_iget(data->obs_mask0, iobs)) {
+                int active_idx = 0;
+                for (int iens = 0; iens < ens_size_msk; iens++) {
+                    if (bool_vector_iget(data->ens_mask, iens)) {
+                        matrix_iset_safe(data->E, iobs, iens,
+                                         matrix_iget(E0, m, active_idx));
+                        active_idx++;
+                    }
+                }
```

Comment:
This looks correct to me, but is there a reason why the `matrix_alloc_column_compressed_copy` that is used in the RML version is not used here? :thinking:

Reply:
Not really sure about that, but this implementation is adapted to be similar to `ies_enkf_data_augment_initialE` (ln204-220)